modules/gateway/peers.go

```diff
@@ -198,10 +204,11 @@ func (g *Gateway) Connect(addr modules.NetAddress) error {
 	var remoteVersion string
 	if err := encoding.ReadObject(conn, &remoteVersion, maxAddrLength); err != nil {
 		return err
-	} else if remoteVersion == "reject" {
-		return errors.New("peer rejected connection")
 	}
 	// decide whether to accept this version
+	if remoteVersion == "reject" {
+		return errors.New("peer rejected connection")
+	}
 	if build.VersionCmp(remoteVersion, "0.3.3") < 0 {
```

Comment:
0333 nodes are actually hardforked off the network at this point, we need to phase out their ability to obtain peers so that they realize they need to upgrade.